android/src/com/mapswithme/maps/editor/data/HoursMinutes.java

```diff
@@ -26,7 +33,21 @@ protected HoursMinutes(Parcel in)
   @Override
   public String toString()
   {
-    return String.format(Locale.US, "%02d:%02d", hours, minutes);
+    final String time = String.format(Locale.US, "%02d:%02d", hours, minutes);
+
+    if (is24HourFormat)
+      return time;
+
+    SimpleDateFormat fmt24 = new SimpleDateFormat("HH:mm");
```

Comment:
I.e. I think it's not a good idea to convert hours and minutes to a string, and then try to parse this string.